velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -331,9 +335,12 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
     } else {
       if (schemaElement.repetition_type ==
           thrift::FieldRepetitionType::REPEATED) {
-        VELOX_CHECK_LE(
-            children.size(), 2, "children size should not be larger than 2");
-        if (children.size() == 1) {
+        if (schema[parentSchemaIdx].converted_type ==
+            thrift::ConvertedType::LIST) {
+          // TODO: the group names need to be checked. According to spec,
+          // https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists
+          // the name of the schema element being 'array' is
+          // also an indication of this is a list type
           // child of LIST
           auto type = TypeFactory<TypeKind::ARRAY>::create(children[0]->type());
```

Comment:
In case of 2-level encoded list, there will not be any children. Will this change handle that?
```
// List<Integer> (nullable list, non-null elements)
optional group my_list (LIST) {
  repeated int32 element;
}
```

Reply:
To clarify, will we segfault here?

Reply:
This 2-level case is segfault both with and without this PR's change. Actually this case would not touch any of the code changes in this PR. So it is separate. We are working on a separate fix for this case

Reply:
@chliang71 thanks for the clarification. It would be nice to throw an error for this case in a separate PR until we support it.